src/plugins/projectGalaxy/components/CustomBlock.vue

```diff
@@ -0,0 +1,272 @@
+<script>
+import Plugin from '../index';
+import { useFlashNotification } from '@/composables/useFlashNotification';
+
+const { notify } = useFlashNotification();
+
+const APP_URL = 'https://galaxy.eco';
+const NO_OAT_IMAGE =
+  'https://snapshotsplugin.s3.us-west-2.amazonaws.com/placeholder.png';
+const IMG_LOGO_GALAXY =
+  'https://d257b89266utxb.cloudfront.net/galaxy/images/avatar/0x4960c283c45e1898c41633c39fb2015167b20dc3-1655711973.png';
+const IMG_ICON_LINK =
+  'https://d257b89266utxb.cloudfront.net/galaxy/images/avatar/0x4960c283c45e1898c41633c39fb2015167b20dc3-1655712057.png';
+const IMG_ICONS = {
+  CLAIMED:
+    'https://d257b89266utxb.cloudfront.net/galaxy/images/avatar/0x4960c283c45e1898c41633c39fb2015167b20dc3-1655712157.png',
+  CLAIMING:
+    'https://d257b89266utxb.cloudfront.net/galaxy/images/avatar/0x4960c283c45e1898c41633c39fb2015167b20dc3-1655712194.png'
+};
+
+const STATES = {
+  NO_OAT: {
+    buttonText: 'No OAT'
+  },
+  WAIT_TO_START: {
+    buttonText: 'Wait to Start'
+  },
+  VOTE_TO_CLAIM: {
+    buttonText: 'Vote to Claim'
+  },
+  CLAIM: {
+    buttonText: 'Claim'
+  },
+  CLAIMING: {
+    buttonText: 'Claiming'
+  },
+  CLAIMED: {
+    buttonText: 'Claimed'
+  },
+  ENDED: {
+    buttonText: 'Campaign Ended'
+  }
+};
+
+// STATES
+const NO_OAT = 'NO_OAT';
+const WAIT_TO_START = 'WAIT_TO_START';
+const VOTE_TO_CLAIM = 'VOTE_TO_CLAIM'; // NOT_VOTED
+const CLAIM = 'CLAIM';
+const CLAIMING = 'CLAIMING';
+const CLAIMED = 'CLAIMED';
+const ENDED = 'ENDED'; // CAMPAIGN ENDED
+
+export default {
+  inject: ['web3', 'notify'],
+  props: ['space', 'proposal', 'results', 'loaded', 'strategies', 'votes'],
+  data() {
+    return {
+      oats: {},
+      loading: false,
+      plugin: new Plugin(),
+      currentState: NO_OAT,
+      currentCampaignUrl: '',
+      currentCampaignId: '',
+      address: '',
+      oatImg: '',
+      imgLogoGalaxy: IMG_LOGO_GALAXY,
+      imgIconLink: IMG_ICON_LINK,
+      imgIcons: IMG_ICONS
+    };
+  },
+  computed: {
+    web3Account() {
+      return this.web3.value.account;
+    },
```

Comment:
We already have this as a computed property in the web3 composable. This is a good example for the advantages of composition API.

```
import { useWeb3 } from '@/composables/useWeb3';
const { web3Account } = useWeb3();
```

This is a good article: https://www.sitepoint.com/vue-composition-api-reusable-components/

Reply:
Thanks for showing the optimization solution, Maybe we could optimize code with options API next version, now we just need to test our code work or not.